back/app/controllers/web_api/v1/permissions_custom_fields_controller.rb

```diff
@@ -2,20 +2,14 @@
 
 class WebApi::V1::PermissionsCustomFieldsController < ApplicationController
   skip_after_action :verify_policy_scoped
-  before_action :set_permissions_custom_field, only: %i[show update destroy]
+  before_action :set_permissions_custom_field, only: %i[show]
 
   def index
     authorize PermissionsCustomField.new(permission: permission)
-    permissions_custom_fields = permission.permissions_custom_fields
-    permissions_custom_fields = paginate permissions_custom_fields
-    permissions_custom_fields = permissions_custom_fields.includes(:custom_field)
 
-    render json: linked_json(
-      permissions_custom_fields,
-      WebApi::V1::PermissionsCustomFieldSerializer,
-      params: jsonapi_serializer_params,
-      include: %i[custom_field]
-    )
+    permissions_custom_fields_service = Permissions::PermissionsCustomFieldsService.new
+    permissions_custom_fields = permissions_custom_fields_service.fields_for_permission(permission, return_hidden: true)
+    render json: WebApi::V1::PermissionsCustomFieldSerializer.new(permissions_custom_fields, params: jsonapi_serializer_params).serializable_hash
```

Comment:
Note: No longer paginated as this was both not needed and also not possible when working with an array of non-persisted objects